tnoodle-ui/src/components/EventsTable.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import CubingIcon from "../CubingIcon";
+
+class EventsTable extends Component {
+  render() {
+    const { events } = this.props;
+    let maxRounds = Math.max.apply(
+      Math,
+      events.map(event => event.rounds.length)
+    );
+    return (
+      <div className="text-center">
+        <table className="table table-striped table-bordered table-hover table-condensed">
+          <thead className="thead-dark">
+            <tr>
+              <th scope="col" />
+              <th scope="col">Event</th>
+              {Array.from(Array(maxRounds), (e, i) => {
+                return (
+                  <th key={i} scope="col">
+                    Groups for Round {i + 1}
+                  </th>
+                );
+              })}
+            </tr>
+          </thead>
+          <tbody>
+            {events.map(event => (
+              <tr>
+                <td>
+                  <CubingIcon key={event.id} event={event.id} />
+                </td>
+                <td className="align-middle">{event.id}</td>
+                {event.rounds.map(round => (
+                  <td className="align-middle">{round.scrambleSetCount}</td>
+                ))}
+
+                {/* Fill empty cells with --- */}
+                {Array.from(Array(maxRounds - event.rounds.length), () => {
```

Comment:
Similar comment as above about `Array.from` vs `Array.map`.

Reply:
Actually this looks like a valid use case for `Array.from` to me! See some random code:
```js
> Array(5).map(x => 10)
[ <5 empty items> ]
> Array.from(Array(5)).map(x => 10)
[ 10, 10, 10, 10, 10 ]
> Array.from({ length: 5 }, x => 10)
[ 10, 10, 10, 10, 10 ]
```
[Here](https://stackoverflow.com/a/5501711)'s a SO answer explaining that.

In this case I'd suggest this:
```js
Array.from({ length: maxRounds - event.rounds.length }, (_, i) =>
  <td key={i} className="align-middle">-- -- --</td>
);
```

Reply:
Used @jonatanklosko's suggestion